lib/xnvme_be_vfio_async.c

```diff
@@ -76,6 +76,9 @@ xnvme_be_vfio_queue_poke(struct xnvme_queue *queue, uint32_t max)
 		rq = __nvme_rq_from_cqe(q->sq, cqe);
 
 		ctx = (struct xnvme_cmd_ctx *)rq->opaque;
+		if (!ctx) {
```

Comment:
This one I am not fond of either, similar to the aio-controlblock.
Something to mull on..